tests/integration/cli/tests/packages/js-script/wasmer.toml

```diff
@@ -12,7 +12,7 @@ module = "saghul/quickjs:quickjs"
 runner = "wasi"
 
 [command.annotations.wasi]
-main_args = ["/src/index.js"]
+main-args = ["/src/index.js"]
```

Comment:
can we have an alias or an error when using the wrong name?

Reply:
I was thinking of adding [`#[serde(deny_unknown_fields)]`](https://serde.rs/container-attrs.html#deny_unknown_fields) to trigger an error when we try to deserialize an unknown field, but then we lose forward-compatibility.

For example, a newer version of wasmer introduces a `enable-threads` field and now older versions of wasmer will error out with `unknown field, "enable-threads", while deserializing WASI metadata` instead of silently ignoring it.

There is also the `#[serde(alias = "...")]` attribute, but it means I'd need to add snake_case aliases to every field in every metadata type, which isn't very practical 😞